### DISCUSSION TITLE
SSG :  The file named with index does not generate the corresponding html ！

### DISCUSSION THREAD

**Discussion by Deja-vuuu** [vercel/next.js#39752] [Help] [answered]
Why is there no index.html?   No index.html is generated. i'm not sure if ssg is working！
 
Why is this happening? I don't see a description of this in the documentation.  
eg:
file structure
<img width="346" alt="image" src="https://user-images.githubusercontent.com/27748682/185603688-6203953c-2bf6-4cb2-b74a-9b0ec451383c.png">
output
<img width="354" alt="image" src="https://user-images.githubusercontent.com/27748682/185603784-ded6fc1e-9c59-4dbf-a6d1-6ee54b981e58.png">

![image](https://user-images.githubusercontent.com/27748682/185611754-3cd9e7da-3f14-4fce-bcbd-038556cbf0df.png)

**Comment by Deja-vuuu** [comment]
<img width="334" alt="image" src="https://user-images.githubusercontent.com/27748682/185826094-87841d60-56fe-4dc0-b6b0-c55822b5bfb9.png">
I see why. It generates the root directory.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

While it is not explicitly said, this bit of the docs explains a bit, https://nextjs.org/docs/routing/introduction:

> The router will automatically route files named index to the root of the directory.
>
> pages/index.js → /
> pages/blog/index.js → /blog

Assuming Static Generation is possible, a path like `/posts/index.tsx`, builds to `/posts.html`. This is also equivalent to using `/posts.tsx`.
